dask_expr/_collection.py

```diff
@@ -151,8 +151,14 @@ def persist(self, fuse=True, combine_similar=True, **kwargs):
         out = self.optimize(combine_similar=combine_similar, fuse=fuse)
         return DaskMethodsMixin.persist(out, **kwargs)
 
-    def compute(self, fuse=True, combine_similar=True, **kwargs):
-        out = self.optimize(combine_similar=combine_similar, fuse=fuse)
+    def compute(
+        self, fuse=True, combine_similar=True, inject_repartition=True, **kwargs
```

Comment:
Hrm, this feels unclean to me.  I would expect to see this in the `optimize` function maybe, or somewhere where it will be called regardless (like `dask.compute(...)`).  I'm also nervous about the special-case. It looks like it was for head/tail?  I'm curious, what was the problem there?

Reply:
It's just unnecessary for head and tail, but you are correct there is no reason not to add it. it will be a single partition anyway.

Yeah I tend to forget about dask.compute, sorry for this. We can't do this in optimise, we definitely don't want to repartition for persist.

Reply:
I'll be happier if we don't have to add a keyword argument, even if we have an unnecessary call.  Maybe `Repartition(Head(...), 1)` can simplify down to just `Head(...)`?


I'll be mildly sad if we're inconsistent between `.compute()` and `dask.compute(...)` but that at least seems innocuous.

Reply:
We can remove the argument.

We can special case in optimise if we want to add support for dask.compute